index.js

```diff
@@ -54,8 +83,23 @@ function getHighestMetascore() {}
  *  getAverageIMDBRating(movies);
  *  //> 7.76
  */
-function getAverageIMDBRating() {}
-
+function getAverageIMDBRating(movies) {
+  if (!movies || !movies.length) {
+   return 0;
+}
+let allRating = [];
+let sum = 0
+let avgRating = 0;
+for(let movie of movies) {
+  allRating.push(movie.imdbRating);
+  for (let i = 0; i < allRating.length; i++){
+    sum = allRating[i]++;
+    avgRating = sum / allRating.length;
+    console.log(sum, "<---");
+  }
+}
```

Comment:
Looks like we're getting a little muddled here.
Nice work trying to console.log what we need to do to solve it and we're not far off.
Our plan should be:
 1 - iterate over the array and add sum;
 2 - divide the sum by the number of items in the array;
 3 - return that number

Our loop is doing an entire extra loop and adding the sum _each time we iterate_.  This means our number is going to be bigger than it needs to be.

With those new steps try again and see if you can crack it.